### ISSUE TITLE
common/malloc.h has commented lines

### ISSUE THREAD

**Issue by bilgili** [ospray/OSPRay#68]
When compiled with g++ 4.8 and "-Wall" parameter, the commented lines are causing "error: multi-line comment [-Werror=comment]" issues. It is not possible to ignore this in the code using gcc ignore pragmas. If not needed, these may be removed ?

**Comment by jeffamstutz** [comment]
This is an issue that I would personally like to see fixed, but it hasn't been a priority for us because it hasn't "been in the way" of us making progress.

A couple of months ago, I tried making the code compile with "-Wall -Werror -pedantic", but (-Wall especially) found THOUSANDS (!) of problems, where almost every file had multiple things wrong in it. At the time, I had decided to not pursue that work through to its finish because a large subset of errors were in our apps/ directory, knowing that I was going to go in and revamp most of our apps at a later time. Now that the apps rework has mostly happened in ```devel```, I will look at this again and see if it's easier to accomplish.

Out of curiosity: are you compiling OSPRay inside another CMake project which has "-Wall" enabled, or did you turn it on manually when compiling OSPRay on its own?

Cheers,
Jeff
